llmfoundry/data/utils.py

```diff
@@ -114,13 +115,32 @@ def get_num_tokens_in_batch(batch: Batch) -> int:
         else:
             input_ids_tokens = batch['input_ids'].numel()
 
+        loss_generating_tokens = None
+        if 'labels' in batch:
+            loss_generating_tokens = int(
+                torch.sum(batch['labels'] != CROSS_ENTROPY_IGNORE_INDEX).item(),
+            )
+
+            # Subtract one for each example in the batch that starts with a non -100,
```

Comment:
@dakinggg I don't think this subtraction isn't necessary. Instead you can just do this:
```python
loss_generating_tokens = int(
                torch.sum(batch['labels'][...,1:] != CROSS_ENTROPY_IGNORE_INDEX).item(),
            )
```

*I just came across this pr while looking into how mosaic's libs handle the gradient accumulation bug recently discussed on x.com

Reply:
ah yeah, that should work too :)